server/src/main/java/org/elasticsearch/index/mapper/flattened/FlattenedFieldMapper.java

```diff
@@ -716,7 +809,9 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
         try {
             // make sure that we don't expand dots in field names while parsing
             context.path().setWithinLeafObject(true);
-            context.doc().addAll(fieldParser.parse(xContentParser));
+            List<IndexableField> fields = fieldParser.parse(xContentParser);
+            context.doc().addAll(fields);
+            addFlattenedDimensionFields(context, fields);
```

Comment:
I wonder if we could change the signature to have the `fieldParser` accept `context` and do the right thing. I feel like it'd save some extra comparisons.

Reply:
@nik9000 Not sure what you mean with do the right thing? Do you mean doing everything in the parse method? I feel like parse should only parse fields rather than, for instance, adding fields to dimensions.